ca/django_ca/forms.py

```diff
@@ -113,7 +113,7 @@ def __init__(self, *args: typing.Any, **kwargs: typing.Any) -> None:
         help_text=_("Password for the private key. If not given, the private key must be unencrypted."),
     )
     expires = forms.DateField(initial=_initial_expires, widget=AdminDateWidget())
-    subject = SubjectField(label="Subject", required=True)
+    subject = SubjectField(label="Subject", required=False)
```

Comment:
Set this to `False`, because the underlying widget already enforces requiring the `Common Name` field. This allows not having to specify all fields for certificates that don't use all of them.

Even though I get where this idea is coming from, I really think this should just be a `CharField`. The unnecessary complexity this introduces in the code-base is not worth the nicer input imo.

Reply:
> Set this to False, because the underlying widget already enforces requiring the Common Name field. This allows not having to specify all fields for certificates that don't use all of them.

Cool - this is an issue I did want to look into already.

> Even though I get where the idea is coming from using a widget here, I really think this should just be a CharField. ...

You are not wrong, but not right either :-D. It's definitely a bad situation, I agree.

On the one hand, the widget does add much nicer input for the vast majority of users, so it is doing something. On the other hand, there is not really a good standard way of expressing certificate subjects either. RFC 4514 has its flaws, and cryptography does not currently sport a parser for that (and my attempt to implement one hasn't been merged).

In any case, I propose you create an issue for that so that we can discuss potential solutions.

Reply:
Will do. Already saw your PR and the concerns @alex expressed – guess we'll see. However for the input mask there's also the option to add https://github.com/elo80ka/django-dynamic-formset to make formsets play nicely. However as far as I'm concerned I think Django admin customization is quite a pain generally.

Usually functionality is preferable over UX for something mostly power-users are gonna use anyways imo – which is the case for a Certificate Authority.

Reply:
> Usually functionality is preferable over UX for something mostly power-users are gonna use anyways imo – which is the case for a Certificate Authority.

Sure it's "power users" using this. But in many cases its system or network administrators who don't know to much about X.509 certificates. One purpose of this project is that these people don't need to fiddle with all the details to get a "good" CA set up going. That's why there even is the web interface, that's why there's profiles: "just select "webserver" and you'll get a decent webserver certificate!"